File: src/PermMissingElement.java

public class PermMissingElement {
    public int solution(int[] A) {
        int SumOfAcctualArray = 0;
        int n = A.length + 1;

        int ArthematicProgression = (n / 2) * (1 + n);

        for (int j = 0; j < A.length; j++) {
            SumOfAcctualArray = SumOfAcctualArray + A[j];
        }
        return ArthematicProgression - SumOfAcctualArray;

    }

    public static void main(String args[]) {
        PermMissingElement PE = new PermMissingElement();
        System.out.println(PE.solution(new int[]{1, 2, 3, 4, 6}));

    }
}

File: src/JS/PermMissingElement.js
function PermMissingElement(A) {
  var SumOfAcctualArray = 0;
  const len = A.length + 1;

  let AP = (len * (len + 1)) / 2;

  for (let i = 0; i < A.length; i++) {
    SumOfAcctualArray += A[i];
  }
  // const SumOfAcctualArray = A.reduce((acc, val) => acc + val, 0);

  const result = AP - SumOfAcctualArray;
  // console.log(result);
  return result;
}
console.log(PermMissingElement([2, 4, 3, 1, 6, 5, 8, 10, 9]));
